include/quaternion.h

```diff
@@ -23,12 +23,15 @@ namespace irr
 namespace core
 {
 
+struct matrix3x4SIMD;
 
 //! Quaternion class for representing rotations.
 /** It provides cheap combinations and avoids gimbal locks.
 Also useful for interpolations. */
 class quaternion : private vectorSIMDf
 {
+	friend struct matrix3x4SIMD;
```

Comment:
don't add this, make quaternion oblivious of matrix3x4SIMD

you can reinterpret_cast quaternion to vectorSIMDf very safely